qml/ProjectList.qml

```diff
@@ -70,232 +110,480 @@ ScrollView
+				if (projectFiles.model.count > 1)
+				{
+					projectFiles.currentRow = 1
+					if (projectFiles.model.get(1).type === "file")
+						projectFiles.executeSelection(1)
+					else
+						projectFiles.currentRow = 0
+				}
+				else
+					projectFiles.currentRow = 0
+			}
+
+			onIsCleanChanged:
 			{
-				Layout.preferredWidth: filesCol.width
-				id: section;
-				sectionName: modelData
-				index:
+				for (var k = 0; k < projectFiles.model.count; k++)
 				{
-					for (var k in sectionRepeater.model)
+					if (projectFiles.model.get(k).path === document.path)
 					{
-						if (sectionRepeater.model[k] === modelData)
-							return k;
+						projectFiles.model.get(k).unsaved = !isClean
+						fileUnsaved(document.path, !isClean)
+						break
 					}
 				}
+			}
+			onDocumentAdded:
+			{
+				projectFiles.updateView()
+			}
+			onFolderAdded:
+			{
+				projectFiles.updateView()
+			}
+		}
 
-				model: sectionModel
-				selManager: sectionRepeater
+		DefaultText
+		{
+			id: folderPath
+			Layout.preferredWidth: filesCol.width
+			Layout.minimumHeight: 18
+			anchors.left: parent.left
+			anchors.leftMargin: 10
+			font.pointSize: appSettings.getFormattedPointSize() - 1
+			color:
+			{
+				return text.indexOf(fileIo.pathFromUrl(projectModel.projectPath)) !== -1 ? "undefined" : "red"
+			}
+			verticalAlignment: Text.AlignVCenter
+		}
 
-				onDocumentSelected: {
-					selManager.selected(doc, groupName);
-				}
+		Splitter
+		{
+			orientation: Qt.Vertical
+			Layout.preferredWidth: filesCol.width
+			Layout.preferredHeight: filesCol.height - projectHeader.height - folderPath.height
 
-				ListModel
-				{
-					id: sectionModel
-				}
+			TableView {
+				id: projectFiles
+				property string currentFolder
+				Layout.minimumHeight: 500
+				clip: true
+				headerDelegate: null
+				itemDelegate: renderDelegate
+				model: ListModel {}
 
-				Connections {
-					id: projectConnection
-					target: codeModel
-					property bool firstLoad: true
-					property bool inErrorAtStart: false
+				Component {
+					id: renderDelegate
+					Item {
+						id: innerItem
 
-					onCompilationError:
-					{
-						if (firstLoad)
+						Rectangle
 						{
-							firstLoad = false
-							if (modelData !== "Contracts")
-								return
-							inErrorAtStart = true
-							for (var k = 0; k < projectModel.listModel.count; k++)
-							{
-								var i = projectModel.listModel.get(k);
-								if (i.isContract)
+							color: "transparent"
+							anchors.verticalCenter: parent.verticalCenter
+							anchors.left: parent.left
+							anchors.leftMargin: 3
+							height: fileName.height
+							Image {
+								source:
 								{
-									sectionModel.append(i)
+									if (styleData.value === "" || styleData.row === -1)
+										return "qrc:/qml/img/unknown.png"
+									if (projectFiles.model.get(styleData.row).type === "folder")
+										return "qrc:/qml/img/folder-icon-64x64.png"
+									else
+										return "qrc:/qml/img/unknown.png"
```

Comment:
![screenshot from 2016-01-18 17 19 15](https://cloud.githubusercontent.com/assets/9685356/12396827/bfe3bbba-be09-11e5-87e2-a9601fc80c52.png)
what about this icon?